ui/src/app/components/workflows/workflow/sensor/sensor.component.spec.ts

```diff
@@ -16,25 +16,100 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 
 import { SensorComponent } from './sensor.component';
+import { provideMockStore } from '@ngrx/store/testing';
+import { DynamicFormPart, DynamicFormParts, FormPart, WorkflowFormPartsModel } from '../../../../models/workflowFormParts.model';
 
 describe('SensorComponent', () => {
-  // let component: SensorComponent;
-  // let fixture: ComponentFixture<SensorComponent>;
-  //
-  // beforeEach(async(() => {
-  //   TestBed.configureTestingModule({
-  //     declarations: [ SensorComponent ]
-  //   })
-  //   .compileComponents();
-  // }));
-  //
-  // beforeEach(() => {
-  //   fixture = TestBed.createComponent(SensorComponent);
-  //   component = fixture.componentInstance;
-  //   fixture.detectChanges();
-  // });
-  //
-  // it('should create', () => {
-  //   expect(component).toBeTruthy();
-  // });
+  let fixture: ComponentFixture<SensorComponent>;
+  let underTest: SensorComponent;
+
+  const initialAppState = {
+    workflows: {
+      workflowFormParts: new WorkflowFormPartsModel(
+        [],
+        new FormPart('switchPartName', 'switchPartProp', true, 'switchPartType', ['optionOne', 'optionTwo']),
+        undefined,
+        undefined,
+        new DynamicFormParts(
+          [
+            new DynamicFormPart('optionOne', [new FormPart('partOne', 'partOne', true, 'partOne')]),
+            new DynamicFormPart('optionTwo', [new FormPart('partTwo', 'partTwo', true, 'partTwo')]),
+          ],
+          [],
+        ),
+      ),
+      workflowAction: {
+        mode: 'mode',
+        workflowData: {
+          sensor: [
+            { property: 'propertyOne', value: 'valueOne' },
+            { property: 'propertyTwo', value: 'valueTwo' },
+            { property: 'switchPartProp', value: 'optionOne' },
+          ],
+        },
+      },
+    },
+  };
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      providers: [provideMockStore({ initialState: initialAppState })],
+      declarations: [SensorComponent],
+    }).compileComponents();
+  }));
```

Comment:
Did you omit the afterViewInit Test on purpose?

Reply:
Done